### PR TITLE
feat: transcoding improvements

### PR DESCRIPTION
**Pull request by zackpollard**
This will bring an array of transcoding improvements to immich, namely:

- [x] Allow transcoding of all videos, regardless of the format they are uploaded in
- [x] Intelligently analyse the uploads with ffprobe to decide if they need to be transcoded (aren't in target format already)
- [x] If a transcoded video exists, the immich web and mobile clients will use that instead of the original for playback

Notes:
- This will reset the users video codec config back to the new default of h264 and audio codec to aac

TODO:
- ~~Check iOS videos play correctly after transcoding #1025 mentions there may be issue with mp4 containers with mp3 audio, possibly switch the defaults to AAC?~~
  - Updated to AAC as the default, confirmed working by @alextran1502

### CODE REVIEW

**Comment by zackpollard** [comment]
This is ready for review and merge once the refactor/system-config PR is merged

**Comment by zackpollard** [comment]
This PR completes another step of #203

File: server/libs/domain/src/system-config/system-config.core.ts [hunk 1]
```diff
@@ -11,9 +11,10 @@ const defaults: SystemConfig = Object.freeze({
   ffmpeg: {
     crf: '23',
     preset: 'ultrafast',
-    targetVideoCodec: 'libx264',
+    targetVideoCodec: 'h264',
```
Comment (alextran1502) [inline server/libs/domain/src/system-config/system-config.core.ts:14]: This option causes quicktime video doesn't have sound anymore

Comment (zackpollard) [reply server/libs/domain/src/system-config/system-config.core.ts:14]: It'll be the MP3 audio, I will make a change to switch that to AAC then you can retest with that

**Pull request merged**